apis/v1alpha1/clientsettingspolicy_types.go

```diff
@@ -0,0 +1,149 @@
+}
+
+// ClientSettingsPolicyConfig contains client settings policy configuration.
+type ClientSettingsPolicyConfig struct {
+	// Body defines the client request body settings.
+	//
+	// +optional
+	Body *ClientBody `json:"body,omitempty"`
+
+	// KeepAlive defines the keep-alive settings.
+	//
+	// +optional
+	KeepAlive *ClientKeepAlive `json:"keepAlive,omitempty"`
+}
+
+// ClientBody contains the settings for the client request body.
+type ClientBody struct {
+	// MaxSize sets the maximum allowed size of the client request body.
+	// If the size in a request exceeds the configured value,
+	// the 413 (Request Entity Too Large) error is returned to the client.
+	// Setting size to 0 disables checking of client request body size.
+	// Default: 1m.
+	//
+	// +optional
+	MaxSize *Size `json:"maxSize,omitempty"`
+
+	// Timeout defines a timeout for reading client request body. The timeout is set only for a period between
+	// two successive read operations, not for the transmission of the whole request body.
+	// If a client does not transmit anything within this time, the request is terminated with the
+	// 408 (Request Time-out) error.
+	// Default: 60s.
+	//
+	// +optional
+	Timeout *Duration `json:"timeout,omitempty"`
+}
+
+// ClientKeepAlive defines the keep-alive settings for clients.
+type ClientKeepAlive struct {
+	// Requests sets the maximum number of requests that can be served through one keep-alive connection.
+	// After the maximum number of requests are made, the connection is closed. Closing connections periodically
+	// is necessary to free per-connection memory allocations. Therefore, using too high maximum number of requests
+	// is not recommended as it can lead to excessive memory usage.
+	// Default: 1000.
+	//
+	// +optional
+	// +kubebuilder:validation:Minimum=0
+	Requests *int32 `json:"requests,omitempty"`
+
+	// Time defines the maximum time during which requests can be processed through one keep-alive connection.
+	// After this time is reached, the connection is closed following the subsequent request processing.
+	// Default: 1h.
+	//
+	// +optional
+	Time *Duration `json:"time,omitempty"`
+
+	// Timeout defines the keep-alive timeouts for clients.
+	//
+	// +optional
+	Timeout *ClientKeepAliveTimeout `json:"timeout,omitempty"`
+}
+
+// ClientKeepAliveTimeout defines the timeouts related to keep-alive client connections.
+type ClientKeepAliveTimeout struct {
+	// Server sets the timeout during which a keep-alive client connection will stay open on the server side.
+	// Setting this value to 0 disables keep-alive client connections.
+	// Default: 75s.
+	//
+	// +optional
+	Server *Duration `json:"server,omitempty"`
+
+	// Header sets the timeout in the "Keep-Alive: timeout=time" response header field.
+	//
+	// +optional
+	Header *Duration `json:"header,omitempty"`
+}
+
+// Duration is a string value representing a duration in time.
+// Duration can be specified in milliseconds (ms), seconds (s), minutes (m), hours (h),
```

Comment:
This is a bit different from the design. 

Duration is no longer a subset of `time.ParseDuration`. This allows an easier mapping to the NGINX time measurement units: https://nginx.org/en/docs/syntax.html

Reply:
do duration rules work the same for all directives in this CRD?

this works:
```
keepalive_timeout 1d 1M;
```

this doesn't:
```
keepalive_timeout 1M 1M;
```

Reply:
Thanks for catching.

Do you know why `keepalive_timeout 1M 1M;` is invalid? The docs say that the "parameters may differ", but do they have to be different?

Reply:
Also, `M` is month which is not included in the regex.

Not sure we need to include days here either.

Reply:
> Do you know why keepalive_timeout 1M 1M; is invalid?

looks like they are of a different time type -- https://github.com/nginx/nginx/blob/e3207a17f084c9eb7905ca0f2cfdb1df088fd165/src/http/ngx_http_core_module.c#L5051-L5063